noxfile.py

```diff
@@ -107,7 +107,7 @@ def _run_integration_tests(session: Session, env: dict[str, str]) -> None:
     netrc = "machine 127.0.0.1 login cachi2-user password cachi2-pass"
     default_env = {"CACHI2_TEST_NETRC_CONTENT": os.getenv("CACHI2_TEST_NETRC_CONTENT", netrc)}
     default_env.update(env)
-    cmd = "pytest --log-cli-level=WARNING tests/integration"
+    cmd = "pytest --log-cli-level=WARNING tests/integration --maxfail=1"
```

Comment:
Haven't we had a discussion about this in the past in a different PR that failing fast might be sometimes counterproductive? I might be misremembering things.
This approach feels to me like computer games where you need to complete a task to get to the next checkpoint only to find out there's a failure waiting for you instead of revealing all failures at once. If the test output is bothersome then the test suite can be filtered to execute only on a subset of tests, but in general having an idea on how many failures one will need to deal with seems like a benefit rather than being something inferior.